proposals/smip-0002-remove-vault-vesting-clff.md

```diff
@@ -0,0 +1,84 @@
+authors:
+  - Lane Rettig <lane@spacemesh.io>
+category: Standard
+type: Core
+status: Draft
+created: 2023-12-26
+---
+
+## Abstract
+
+This SMIP removes the 25% vesting cliff one year post-genesis from the coins in the genesis vesting vaults with a slight change to the interpretation of the vault account parameters. After this change, the coins in these vaults vest linearly over a three-year period, from one year post-genesis to four years post-genesis.
+
+## Scope
+
+This proposal requires a slight change to the precompile/genesis VM code in go-spacemesh. One parameter in the economics library (https://github.com/spacemeshos/economics/) will also need to change.
+
+It has no additional impact on and requires no support from dapp developers, miners, etc.
+
+## Stakeholders and Reviewers
+
+- Spacemesh core contributors (who have already reviewed and approved this proposed change)
+- Spacemesh development company investors (who have already reviewed and approved this proposed change)
+- all Spacemesh miners, hodlers, and community members (since there is a short-term economic impact years 1-4) are given the chance to review this proposal and comment on it here, in the community Discord, and in the upcoming town hall (details TBA)
+
+## Motivation
+
+See https://spacemesh.io/blog/smoothing-spacemesh-economics/.
+
+## Alternatives Considered
+
+The only alternative considered was not making this change and maintaining the current vesting schedule.
+
+## Specification
+
+The `InitialUnlockAmount` parameter passed to the `Vault` template when a vault account is instantiated (see [this code snippet](https://github.com/spacemeshos/go-spacemesh/blob/eb55894737514b456bbd4af4a41178639a6bda15/genvm/templates/vault/vault.go#L26-L34)) must be ignored when calculating the available balance in such an account.
+
+For clarity, the old algorithm was:
+
+`AVAILABLE_BALANCE := INITIAL_UNLOCK_AMOUNT + (TOTAL_AMOUNT - INITIAL_UNLOCK_AMOUNT) * (LAYER - VESTING_START) / (VESTING_END - VESTING_START)`
+
+This proposal changes the algorithm to:
+
+`AVAILABLE_BALANCE := TOTAL_AMOUNT * (LAYER - VESTING_START) / (VESTING_END - VESTING_START)`
+
+## Rationale
+
+This is the simplest possible change to effect the desired result. It doesn't change the state of any existing accounts, and the address of the existing accounts also does not change.
+
+## Backwards Compatibility
+
+This proposal changes the interpretation of the arguments passed to the genesis vault accounts. As long as this change is implemented well before vesting begins (one year post genesis), no special backwards-compatibility handling is required. Additionally, no new genesis vault accounts will ever be created on mainnet, so no backwards-compatibility handling is required for this account type.
+
+If this proposal is implemented, in general, developers need to be aware that the `InitialUnlockAmount` vault argument is ignored and should be ignored in all Spacemesh infrastructure and applications.
```

Comment:
there is another alternative. vault template definition can be changed so that there is no InitialUnlockAmount at all.  it will require zeroing out old vaults, and generating new addresses using new spawn arguments. the advantage is to avoid confusion in the future about removed parameter, but it will require migration.  

the alternative that i described feels less hacky, but i don't have simpler option (which is described in the smip) too.

Reply:
Thanks for the feedback

I merged this SMIP in its current form. Let's move the discussion to https://github.com/smfoundation/smips/discussions/2. Will link this thread there.